ml-agents/mlagents/trainers/ppo/trainer.py

```diff
@@ -253,14 +253,12 @@ def add_policy(
                     self.__class__.__name__
                 )
             )
-        if not isinstance(policy, NNPolicy):
-            raise RuntimeError("Non-NNPolicy passed to PPOTrainer.add_policy()")
-        self.policy = policy
+        self.policy = self.create_policy(brain_parameters)
         self.optimizer = PPOOptimizer(self.policy, self.trainer_parameters)
```

Comment:
Almost want to say the add_policy method should go in `rl_trainer` with the Optimizer as an object (e.g. `self.optimizer_class = PPOOptimizer` and then in the method `self.optimizer = self.optimizer_class(self.policy)`. Since it's duplicated here and in SAC.